### PR TITLE
Added unit tests for WebApi ServiceCollection and AuthenticationBuilder extensions. Minor fixes.

### PR DESCRIPTION
**Pull request by pmaytak**
Added unit tests for
- WebApiAuthenticationBuilderExtensions
- WebApiServiceCollectionExtensions

Note: I put them into one test class as the tests are pretty much the same.

### CODE REVIEW

File: tests/Microsoft.Identity.Web.Test.Common/TestConstants.cs [hunk 1]
```diff
@@ -81,5 +81,16 @@ public static class TestConstants
         public const string ConfidentialClientKeyVaultUri = "https://buildautomation.vault.azure.net/secrets/AzureADIdentityDivisionTestAgentSecret/";
         public const string ConfidentialClientId = "16dab2ba-145d-4b1b-8569-bf4b9aed4dc8";
         public const string ConfidentialClientLabTenant = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+
+        //This value is only for testing purposes. It is for a certificate that is not used for anything other than running tests
+        public const string certificateX5c = @"MIIDHzCCAgegAwIBAgIQM6NFYNBJ9rdOiK+C91ZzFDANBgkqhkiG9w0BAQsFADAgMR4wHAYDVQQDExVBQ1MyQ2xpZW50Q2VydGlmaWNhdGUwHhcNMTIwNTIyMj
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Web.Test.Common/TestConstants.cs:86]: I got this from [MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/aef655d4575d5c4849de9f8a1df0ff7ac38affe6/tests/Microsoft.Identity.Test.Common/TestConstants.cs#L117). @jennyf19 let me know if there's a better way to get cert data or this is fine.

File: src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -121,6 +122,8 @@ public static class WebApiServiceCollectionExtensions
 
             services.Configure<JwtBearerOptions>(jwtBearerScheme, options =>
             {
+                options.Events ??= new JwtBearerEvents();
+
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:126]: nit: spacing

Comment (pmaytak) [reply src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:126]: What do you mean?

File: tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+        }
+
+        private IConfigurationSection GetConfigSection(string configSectionName)
+        {
+            var configAsDictionary = new Dictionary<string, string>()
+            {
+                { configSectionName, null },
+                { $"{configSectionName}:Instance", TestConstants.AadInstance },
+                { $"{configSectionName}:TenantId", TestConstants.TenantIdAsGuid },
+                { $"{configSectionName}:ClientId", TestConstants.TenantIdAsGuid }
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs:281]: >TenantId [](start = 65, length = 8)

ClientId?

File: tests/Microsoft.Identity.Web.Test.Common/TestConstants.cs (hunk 1 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Web.Test.Common/TestConstants.cs:86]: yes...this should be great.

---
In reply to: [406834661](https://github.com/AzureAD/microsoft-identity-web/pull/93#discussion_r406834661) [](ancestors = 406834661)

File: tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs (hunk 3 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/WebApiExtensionsTests.cs:281]: I just needed any valid Guid here but I added a TestConstants.ClientId for clarity.

File: src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:126]: the space between ??=

Comment (pmaytak) [reply src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:126]: There shouldn't be one. It's [compound operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator).

Comment (bgavrilMS) [reply src/Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs:126]: Hah, never used this operator before. It's nice to see smth new :)

**Pull request merged**